src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3021,6 +3021,11 @@ object Types {
 
   object ErrorType extends ErrorType
 
+  /* Type used to track Select nodes that could not resolve a member and their qualifier is a scala.Dynamic. */
+  abstract class TryDynamicCallType extends UncachedGroundType with ValueType
+
+  object TryDynamicCallType extends TryDynamicCallType
+
```

Comment:
Actually, now that I think about it, it should be `object TryDynamicCallType extends ErrorType`.